File: Week-1-April-1st-7th/Maximum-Subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currentSum = 0, ans = INT_MIN;
        for(auto x : nums){
            currentSum += x;
            ans = max(ans, currentSum);
            currentSum = max(currentSum, 0);
        }
        return ans;
    }
};

File: Week-1-April-1st-7th/Maximum-Subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_sum = 0
        ans = float("-inf")
        for x in nums:
            current_sum += x
            ans = max(ans, current_sum)
            current_sum = max(current_sum, 0)
        return ans
